src/sacn/sockets.c

```diff
@@ -321,14 +321,17 @@ etcpal_error_t sacn_add_receiver_socket(sacn_thread_id_t thread_id, etcpal_iptyp
   return res;
 }
 
-void sacn_remove_receiver_socket(sacn_thread_id_t thread_id, etcpal_socket_t socket, bool close_now)
+void sacn_remove_receiver_socket(sacn_thread_id_t thread_id, etcpal_socket_t *socket, bool close_now)
 {
-  SACN_ASSERT(socket != ETCPAL_SOCKET_INVALID);
+  SACN_ASSERT(socket != NULL);
+  SACN_ASSERT(*socket != ETCPAL_SOCKET_INVALID);
 
   SacnRecvThreadContext* context = get_recv_thread_context(thread_id);
   SACN_ASSERT(context);
 
-  cleanup_socket(context, socket, close_now);
+  cleanup_socket(context, *socket, close_now);
```

Comment:
Normally, I don't try to rely on asserts to guarantee validity.  I would typically check the validity of context & socket at runtime.  Only with good analysis would I consider doing this in a function that has critical performance needs.

Reply:
Assert should be used when a function has preconditions that would only be violated if there was a bug in the library. I believe this qualifies. We should fail early and hard in the presence of bugs and not use runtime checks to cover them up.

Reply:
When I wrote this part, I was also faced with the decision of using assert or runtime checks, and I went asserts for the reasons Sam describes. I felt that this way would provide stricter error checking and I couldn't think of any drawbacks. Are you okay with keeping this as is @nwagner ?

Reply:
We talked offline and decided to leave this as-is.